core/relayMsgs.go

```diff
@@ -70,23 +77,38 @@ func (r *RelayMsgs) Send(src, dst Chain) {
 
 		if r.IsMaxTx(msgLen, txSize) {
 			// Submit the transactions to src chain and update its status
-			r.Succeeded = r.Succeeded && SendCheckMsgs(src, msgs)
-
+			msgIDs, err := src.SendMsgs(msgs)
+			if err != nil {
+				logger.Error("failed to send msgs", err, "msgs", msgs)
+			}
+			r.Succeeded = r.Succeeded && (err == nil)
+			for i := range msgs {
+				srcMsgIDs[i+maxTxCount] = msgIDs[i]
+			}
```

Comment:
if `err` is not nil, `msgIDs` becomes undefined and we shouldn't use it.
In that case, please just skip this loop. In any case it is necessary to update `maxTxCount`.
```
if err == nil {
	for i := range msgs {
		srcMsgIDs[i+maxTxCount] = msgIDs[i]
	}
}
maxTxCount += len(msgs)
```

Reply:
You need apply this change to the three other locations, too.